shanoir-ng-datasets/src/main/java/org/shanoir/ng/examination/service/ExaminationServiceImpl.java

```diff
@@ -121,6 +133,21 @@ public void deleteFromRabbit(Examination exam) throws EntityNotFoundException {
 			return examinationRepository.findByPreclinicalAndStudyIdIn(preclinical, studyIds, pageable);
 		}
 	}
+	
+	@Override
+	public Page<Examination> findPage(final Pageable pageable, String patientName) {
+		if (KeycloakUtil.getTokenRoles().contains("ROLE_ADMIN")) {
+			if (StringUtils.isNotEmpty(patientName) && patientName.length() <= 64) {
+				return examinationRepository.findAllBySubjectName("%" + patientName + "%", pageable);
+			} else {
+				return examinationRepository.findAll(pageable);
+			}
+		} else {
+			Long userId = KeycloakUtil.getTokenUserId();
```

Comment:
So if the user is not an admin, he can't search a subject name ?

Reply:
I have to fix that, thank you, I forgot about it

Reply:
fixed

Reply:
just for the notes: I tried a cleaner solution on not-using @query (native query) and changed the type of Long subjectId to Subject subject and I could avoid the usage of @query, BUT this introduced other issues afterwards with Gson in the Unittests, that I was not able to solve within 1 hour, so I stopped to one day finish this PR. The problem lies, that Gson has even be introduced, what should not have been done. Gson annotations are different to the Jackson ones, and to fix my issue, I would have to double annotate Java classes for json with both annotation types, what would be even more confusing than anything else. One solution would be to clean up all Gson usage in the Unit tests, what is quite a task.